Snowflake.Data/Logger/SFLoggerFactory.cs

```diff
@@ -1,51 +1,81 @@
-﻿/*
+/*
  * Copyright (c) 2012-2019 Snowflake Computing Inc. All rights reserved.
  */
 
-using log4net;
+using Microsoft.Extensions.Logging;
+using System.IO;
 
 namespace Snowflake.Data.Log
 {
-    class SFLoggerFactory
+    public class SFLoggerFactory
     {
-        private static bool isLoggerEnabled = true;
+        private static bool s_isCustomLoggerEnabled = false;
 
-        private static SFLogger logger = null;
+        private static bool s_isSFLoggerEnabled = true;
 
-        private SFLoggerFactory()
+        private static ILogger s_customLogger = null;
+
+        public static void DisableSFLogger()
+        {
+            s_isSFLoggerEnabled = false;
+        }
+
+        public static void EnableSFLogger()
+        {
+            s_isSFLoggerEnabled = true;
+        }
+
+        public static void DisableCustomLogger()
         {
+            s_isCustomLoggerEnabled = false;
         }
 
-        public static void disableLogger()
+        public static void EnableCustomLogger()

```

Comment:
Let's say that that at the beginning customer logger was disabled.
Let's say that some class has a static logger eq. `private static readonly SFLogger logger = SFLoggerFactory.GetLogger<BasicAuthenticator>();`
then the logger would have a value of `new SFLoggerPair(..., ILoggerEmptyImpl();)`
Let's say then the customer would call `EnableCustomLogger()`.
The logger obtained in the first line would not be modified.
Am I right?
I think we need to reconfigure all the logger when `EnableCustomLogger()`, switching this field value is not enough.